src/QsCompiler/SyntaxProcessor/ExpressionVerification.fs

```diff
@@ -446,8 +445,7 @@ type QsExpression with
             let resolveSlicingRange start step end_ =
                 let integerExpr ex =
                     let ex = resolve context ex
-                    inference.Constrain(ResolvedType.New Int .> ex.ResolvedType) |> List.iter diagnose
-                    ex
+                    { ex with ResolvedType = inference.Resolve ex.ResolvedType }
```

Comment:
Can you rename this function since it isn't related to integers anymore?

Reply:
Oh yah I guess this function's purpose changed. Renamed to `toResolvedExpr`.